From cb1fd2204660b7a9432e043829c7db30cf288265 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Thu, 9 Feb 2012 10:36:56 +0100
Subject: [PATCH] install: fix incorrect 'Access denied' message with a
 non-existent unit

With "systemctl is-enabled non-existent.service"
_UNIT_FILE_STATE_INVALID (-1) was wrongly interpreted as -errno.
Return -ENOENT in this case.

https://bugzilla.redhat.com/show_bug.cgi?id=766579
(cherry picked from commit 81006b8ad8cfdbfdb418fc99918ee5c33eb5b35a)
---
 src/install.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/install.c b/src/install.c
index 1fb1f9d..174d79b 100644
--- a/src/install.c
+++ b/src/install.c
@@ -1571,10 +1571,10 @@ UnitFileState unit_file_get_state(
                 }
 
                 if (lstat(path, &st) < 0) {
+                        r = -errno;
                         if (errno == ENOENT)
                                 continue;
 
-                        r = -errno;
                         goto finish;
                 }
 
